### PR TITLE
Add plotting script

### PR DESCRIPTION
**Pull request by elynnwu**
## Purpose

Including the plotting script used as part of our debug process. More example usage will come in separate PRs.

## Code changes:

- add `plot_pcolormesh_cube` that is able to plot: absolute pace output at a given time step and height, or subtracting the first time step, or subtracting the given fortran output

## Checklist
Before submitting this PR, please make sure:

- [x] You have followed the coding standards guidelines established at [Code Review Checklist](https://paper.dropbox.com/doc/Code-Review-Checklist--BD7zigBMAhMZAPkeNENeuU2UAg-IlsYffZgTwyKEylty7NhY).

### CODE REVIEW

File: driver/examples/plot_pcolormesh_cube.py [hunk 1]
```diff
@@ -0,0 +1,202 @@
...
+        klevel: index number in the k-axis to be read
+    """
+    ts_size = len(
+        xr.open_dataset(path + "/atmos_custom_fine_inst.tile1.nc", decode_times=False)[
+            "time"
+        ]
+    )
+    fortran_data = np.zeros((ts_size, total_ranks, cn, cn))
+    for rank in range(total_ranks):
+        f = xr.open_dataset(
```
[the remark this answers is not in the document]
Comment (elynnwu) [reply driver/examples/plot_pcolormesh_cube.py:117]: Good idea!

File: driver/examples/plot_pcolormesh_cube.py [hunk 2]
```diff
@@ -0,0 +1,202 @@
...
+            path + "/atmos_custom_fine_inst.tile" + str(rank + 1) + ".nc",
+            decode_times=False,
+        )
+        fortran_data[:, rank, :, :] = f[var][:, klevel, :, :]
+        f.close()
+    rotated_fortran_data = np.zeros((ts_size, total_ranks, cn, cn))
```
[the remark this answers is not in the document]
Comment (elynnwu) [reply driver/examples/plot_pcolormesh_cube.py:123]: Moved for loop over time inside contextmanager, and do transpose in place.

File: driver/examples/plot_pcolormesh_cube.py [hunk 3]
```diff
@@ -0,0 +1,202 @@
...
+        temp = fortran_data[t, :, :, :]
+        for rank in range(total_ranks):
+            rotated_fortran_data[t, rank, :, :] = temp[rank, :, :].T
+    return rotated_fortran_data
+
+
+args = parse_args()
+if args.fortran_data_path is not None:
+    if args.fortran_var is None:
+        raise ValueError(
```
[the remark this answers is not in the document]
Comment (elynnwu) [reply driver/examples/plot_pcolormesh_cube.py:134]: Added this check

File: driver/examples/plot_pcolormesh_cube.py [hunk 4]
```diff
@@ -0,0 +1,202 @@
...
+            "You must specify the variable name to be subtracted in Fortran data."
+        )
+    fortran = gather_fortran_data_at_klevel(
+        args.zarr_output, 6, args.size, args.fortran_var, args.zlevel
```
[the remark this answers is not in the document]
Comment (elynnwu) [reply driver/examples/plot_pcolormesh_cube.py:138]: Actually I miswrote my description, this is the total number of tiles, which will always be 6 for us. Removed this as argument into the function.

File: driver/examples/plot_pcolormesh_cube.py [hunk 5]
```diff
@@ -112,91 +114,114 @@ def gather_fortran_data_at_klevel(
...
     else:
-        h = pcolormesh_cube(
-            python_lat,
-            python_lon,
-            plotted_data,
-            ax=ax,
+        # when plotting through sarus or docker
+        # we expect pace output mounted at /pace
+        ds = xr.open_zarr(
+            store=zarr.DirectoryStore(path="/pace/output.zarr"), consolidated=False
```
Comment (twicki) [inline driver/examples/plot_pcolormesh_cube.py:152]: we could simplify the logic here and make this the default for the argument. But if you want it here explicitly I'm fine with it

File: driver/examples/plot_pcolormesh_cube.py [hunk 6]
```diff
@@ -112,91 +114,114 @@ def gather_fortran_data_at_klevel(
...
+            xycoords="axes fraction",
+            ha="right",
+            va="center",
+            fontsize=8,
+        )
+        plt.tight_layout()
+        if args.zarr_output is not None:
+            save_path = ""
+        else:
+            save_path = "/work/"
```
Comment (twicki) [inline driver/examples/plot_pcolormesh_cube.py:222]: should this not be `/pace`?

Comment (twicki) [reply driver/examples/plot_pcolormesh_cube.py:222]: I understand now

**Review by twicki — APPROVED** [review]
lgtm

**Pull request merged**